sources/packages/web/src/components/generic/HeaderNavigator.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <slot name="title">
+    <div v-if="routeLocation" class="header-title">
+      <a @click="goBack()">
+        <font-awesome-icon :icon="['fas', 'arrow-left']" class="mr-2" />
+        {{ title }}</a
+      >
+    </div>
+    <div v-else class="header-title">
+      {{ title }}
+    </div>
+  </slot>
+  <v-row>
+    <v-col>
+      <slot name="subTitle">
+        <div class="header-sub-title">
+          {{ subTitle }}
+        </div>
+      </slot>
+    </v-col>
+    <v-col>
+      <div class="float-right ml-2 header-button">
+        <slot name="buttons"> </slot>
+      </div>
+    </v-col>
+  </v-row>
+</template>
+<script lang="ts">
+import { useRouter } from "vue-router";
+
+export default {
+  props: {
+    title: {
+      type: String,
+    },
+    subTitle: {
+      type: String,
+    },
+    routeLocation: {
+      type: Object,
+    },
+  },
+
+  setup(props: any) {
+    const router = useRouter();
+
+    const goBack = () => {
+      if (props?.routeLocation?.routeParams) {
```

Comment:
I believe that this entire if/else block can be replaced by the below code, doesn't? Am I missing something?
```javascript
router.push(props.routeLocation);
```

Reply:
I tried the code above but no response when I click the back arrow icon.

The one below works.
router.push({
        name: props?.routeLocation?.routeName,
        params: props?.routeLocation?.routeParams,
      });

Reply:
I removed my previous comment and refactored to get it working according to your recommendation. Thanks @andrewsignori-aot